Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp

```diff
@@ -1369,21 +1369,26 @@ namespace AZ::Data
         {
             AZ_Assert(itr->second->GetContainerAssetId() == assetId,
                 "Asset container is incorrectly associated with the asset being destroyed.");
-            itr->second->ClearRootAsset();
 
-            // Only remove owned asset containers if they aren't currently loading.
-            // If they *are* currently loading, removing them could cause dependent asset loads that were triggered to
-            // remain in a perpetual loading state.  Instead, leave the containers for now, they will get removed during
-            // the OnAssetContainerReady callback.
-            if (!itr->second->IsLoading())
+            // Make sure the asset instance we're releasing is the same as the one for the container
+            // Sometimes old references (from before a reload) are released which should not cancel newer loads
+            if (itr->second->GetRootAsset()->GetCreationToken() == asset->GetCreationToken())
             {
-                m_ownedAssetContainers.erase(itr->second);
-                itr = m_ownedAssetContainerLookup.erase(itr);
-            }
-            else
-            {
-                ++itr;
+                itr->second->ClearRootAsset();
+
+                // Only remove owned asset containers if they aren't currently loading.
+                // If they *are* currently loading, removing them could cause dependent asset loads that were triggered to
+                // remain in a perpetual loading state.  Instead, leave the containers for now, they will get removed during
+                // the OnAssetContainerReady callback.
+                if (!itr->second->IsLoading())
+                {
+                    m_ownedAssetContainers.erase(itr->second);
```

Comment:
Is there any danger of the iter not being found in m_ownedAssetContainers or m_ownedAssetContainers ie we update one but not the other?

Reply:
m_ownedAssetContainerLookup exists for the purpose of being able to find the container in m_ownedAssetContainers, so they should be in sync.  If for some reason its missing it shouldn't really cause a problem, erase that doesn't find anything just does nothing.